README.md

```diff
@@ -242,7 +242,6 @@ The following targets are available. Each may be run with `make <target>`.
 | `gobuild`              | Runs a normal `go build` of the agent and stores the binary in `./out/amazon-ecs-agent` |
 | `static`               | Runs `go build` to produce a static binary in `./out/amazon-ecs-agent` |
 | `test`                 | Runs all unit tests using `go test` |
-| `test-in-docker`       | Runs all tests inside a Docker container |
```

Comment:
Why is `test-in-docker` going away, and can you add a `test-in-docker-init` to the readme?

Reply:
This was referencing an agent target that no longer exists, decided to remove it as to avoid confusion with the init one.

I thought about how to best approach updating the README. My thinking is, since make targets and ways we are building are a bit in flux and not cleaned up yet, that we should update the README at a later point with all of the updated targets we're going to keep, and updated instructions for the dockerless buillds. Otherwise, might risk adding information that will quickly become incorrect and might forget to update it